engine/crates/engine-v2/schema/src/builder/mod.rs

```diff
@@ -469,21 +451,26 @@ impl From<federated_graph::Object> for Object {
             name: object.name.into(),
             description: None,
             interfaces: object.implements_interfaces.into_iter().map(Into::into).collect(),
-            composed_directives: object.composed_directives.into_iter().map(Into::into).collect(),
+            composed_directives: convert_range(object.composed_directives),
             cache_config: Default::default(),
         }
     }
 }
 
 impl From<federated_graph::Directive> for Directive {
     fn from(directive: federated_graph::Directive) -> Self {
-        Directive {
-            name: directive.name.into(),
-            arguments: directive
-                .arguments
-                .into_iter()
-                .map(|(id, value)| (id.into(), value.into()))
-                .collect(),
+        match directive {
+            federated_graph::Directive::Inaccessible => Directive::Inaccessible,
```

Comment:
I'm not a fan of keeping `Inaccessible`, but removing those fields would make all id conversions a lot more complex, so well, let's keep it.

But this isn't enough, an inaccessible field will be exposed through introspection _and_ be accessible through queries here.